### PR TITLE
fix: handle more ambiguous spec edge cases for simple jinja2 statements and in test deps

### LINKED ISSUE #307: Error converting a recipe with a set `python_min` variable
The bug was originally reported at https://github.com/hadim/feedrattler/issues/4.

When trying to convert https://github.com/conda-forge/pyhf-feedstock/blob/5f1ecaec210f588db912f57823a3d142a7c50da4/recipe/meta.yaml with `pixi exec conda-recipe-manager convert meta.yaml`, the command generates:

```yaml
schema_version: 1

context:
...
  python_min: 3.7

...

requirements:
  host:
    - python 3.7.*
```

instead of 

```yaml
schema_version: 1

context:
...
  python_min: 3.7

...

requirements:
  host:
    - python ${{ python_min }}.*
  
```

### PR DESCRIPTION
**Pull request by beckermr**
### Description

This PR fixes edge cases where the conversion function does not add `.*` to specs like `python {{ python_min }}` when `python_min` does not have a corresponding set statement.

It also fixes cases where ambiguous test deps were not converted.

closes #307

### CODE REVIEW

File: conda_recipe_manager/parser/_types.py [hunk 1]
```diff
@@ -223,6 +226,7 @@ class Regex:
 
     ## Jinja regular expressions ##
     JINJA_V0_SUB: Final[re.Pattern[str]] = re.compile(r"{{\s*" + _JINJA_VAR_FUNCTION_PATTERN + r"\s*}}")
+
```
Comment (beckermr) [inline conda_recipe_manager/parser/_types.py]:
```suggestion
```

**Review by schuylermartin45 — APPROVED** [review]
Looks pretty straight forward to me, always happy to see the community contribute!
